File: src/3248_SnakeInMatrix/3248.go
func finalPositionOfSnake(n int, commands []string) int {
    i := 0 
    j := 0 
    for _, cmd := range commands {
        switch cmd {
        case "UP":    i -= 1
        case "DOWN":  i += 1
        case "LEFT":  j -= 1
        case "RIGHT": j += 1
        }
    }

    return (i * n) + j
}

File: src/3248_SnakeInMatrix/3248.scala
object Solution {
    def finalPositionOfSnake(n: Int, commands: List[String]): Int = {
        val d = commands.map((cmd) => {
            cmd match {
                case "UP"    => (-1, 0)
                case "DOWN"  => (1, 0)
                case "LEFT"  => (0, -1)
                case "RIGHT" => (0, 1)
                case _ => (0,0)
            }
        }).reduce( (a: (Int,Int) ,b: (Int, Int)) => (a(0)+b(0), a(1)+b(1)))
        d(0) * n + d(1)
        
    }
}